src/RevMem.cc

```diff
@@ -13,45 +13,31 @@
 #include <cstring>
 #include <iomanip>
 #include <memory>
+#include <new>
 #include <utility>
 
 namespace SST::RevCPU {
 
 using MemSegment = RevMem::MemSegment;
 
-RevMem::RevMem( uint64_t memSize, RevOpts* opts, RevMemCtrl* ctrl, SST::Output* output )
-  : memSize( memSize ), opts( opts ), ctrl( ctrl ), output( output ) {
-  // Note: this constructor assumes the use of the memHierarchy backend
-  pageSize  = 262144;  //Page Size (in Bytes)
-  addrShift = lg( pageSize );
-  nextPage  = 0;
-
-  // We initialize StackTop to the size of memory minus 1024 bytes
-  // This allocates 1024 bytes for program header information to contain
-  // the ARGC and ARGV information
-  stacktop  = ( _REVMEM_BASE_ + memSize ) - 1024;
+// We initialize StackTop to the size of memory minus 1024 bytes
+// This allocates 1024 bytes for program header information to contain
+// the ARGC and ARGV information
 
-  // Add the 1024 bytes for the program header information
-  AddMemSegAt( stacktop, 1024 );
+// Note: this constructor assumes the use of the memHierarchy backend
+RevMem::RevMem( uint64_t memSize, RevOpts* opts, RevMemCtrl* ctrl, SST::Output* output )
+  : memSize( memSize ), opts( opts ), ctrl( ctrl ), output( output ), pageSize( 262144 ), addrShift( uint32_t( lg( pageSize ) ) ),
```

Comment:
There were problems with pointer types like `char*` and I fixed them and condensed the code by putting it in the initialization list rather than assigning to members inside the constructor body. The compiler can usually optimize away double-initializations, but technically, all members of a class are "initialized" in a constructor initialization list rather than assignments made in the body of the constructor.